### PR TITLE
First draft of tc_traffic_calming as per #1

### PR DESCRIPTION
**Pull request by layik**
Please comment and request changes!

### CODE REVIEW

**Comment by layik** [comment]
I expected the fail and will see what's the convention for packages not on CRAN (osmextract in this case)

File: DESCRIPTION [hunk 1]
```diff
@@ -39,8 +40,10 @@ Suggests:
     ComplexUpset,
     ggplot2,
     scales,
-    curl
+    curl,
+    sf
 Remotes:
-    krassowski/complex-upset
+    krassowski/complex-upset,
+    itsleeds/osmextract
```
Comment (Robinlovelace) [inline DESCRIPTION:47]: Remotes is how you specify packages not on CRAN @layik. Let's see if this passes the tests, hope so!

**Review by Robinlovelace — APPROVED** [review]
Great work, ready to merge assuming it passes the tests.

**Comment by layik** [comment]
Let me deal with the warnings!

**Comment by layik** [comment]
@Robinlovelace `sf` is the culprit on the Xenial distros. Let me know if you need me to have a look. The notes are fine even with CRAN.

**Comment by Robinlovelace** [comment]
I think this block from the equivalent `sf` .yaml file will solve it:

```
      - name: Install system dependencies
        if: runner.os == 'Linux'
        env:
          RHUB_PLATFORM: linux-x86_64-ubuntu-gcc
        run: |
          Rscript -e "remotes::install_github('r-hub/sysreqs')"
          sysreqs=$(Rscript -e "cat(sysreqs::sysreq_commands('DESCRIPTION'))")
          sudo -s eval "$sysreqs"
          # install spatial dependencies
          sudo add-apt-repository ppa:ubuntugis/ubuntugis-unstable
          sudo apt update
          sudo apt install \
            libudunits2-dev \
            libgdal-dev \
            libgeos-dev \
            libproj-dev
```

https://github.com/r-spatial/sf/blob/master/.github/workflows/R-CMD-check.yaml#L65-L81

**Comment by layik** [comment]
OK all seems to be good now @Robinlovelace. Over to you to see if the function does what you were thinking in #1 but happy to tweak, improve or rewrite completely.

**Comment by Robinlovelace** [comment]
Great work.

**Pull request merged**